ncm-spma/src/main/perl/spma/ips.pm

```diff
@@ -411,27 +443,42 @@ sub run_pkg_command
 
 #
 # Get pkg@ver keys from hash where ver is actually the value.
+# Can be given an existing array to amend.
 #
 sub pkg_keys
 {
-    my ($self, $hash) = @_;
+    my ($self, $hash, $keylist) = @_;
+
+    $keylist ||= [];
 
-    my @keys;
     for my $pkg (keys %$hash) {
```

Comment:
This doesn't actually make a difference, and unfortunately if I were to do this now it would change the output of the cache file making our existing Solaris servers think there is a pending package update when really there is not.